pinot-common/src/main/java/com/linkedin/pinot/common/utils/helix/HelixHelper.java

```diff
@@ -410,4 +410,22 @@ public IdealState apply(IdealState idealState) {
     }
     return enabledInstances;
   }
+
+  public static List<String> getInstancesWithTag(HelixAdmin helixAdmin, String helixClusterName,
```

Comment:
lets get all instanceConfigs in one shot. that should not throw any exceptions and is more efficient.

Reply:
I don't think I understand. That's exactly what this method is doing

Reply:
Understood what you're referring to from the conversation offline. Will change it accordingly

Reply:
@kishoreg made the changes to get all instance configs in one shot